src/logging.c

```diff
@@ -376,32 +409,18 @@ rstrnt_upload_log (const RstrntTask    *task,
     int msgc = 0;
     const char *contents;
     size_t length;
-    GFile *data_file = NULL;
-    gchar *log_path = NULL;
+    const gchar *log_path = NULL;
+    RstrntLogData *log_data = NULL;
 
     manager = rstrnt_log_manager_get_instance ();
     data = rstrnt_log_manager_get_task_data (manager, task, &error);
 
     g_return_if_fail (NULL != data);
 
-    switch (type)
-    {
-        case RSTRNT_LOG_TYPE_TASK:
-        {
-            data_file = data->task_log_data->file;
-            log_path = LOG_PATH_TASK;
-        }
-        break;
-
-        case RSTRNT_LOG_TYPE_HARNESS:
-        {
-            data_file = data->harness_log_data->file;
-            log_path = LOG_PATH_HARNESS;
-        }
-        break;
-    }
+    log_path = rstrnt_log_type_get_path (type);
+    log_data = rstrnt_task_log_get_data (data, type);
 
-    path = g_file_get_path (data_file);
+    path = g_file_get_path (log_data->file);
```

Comment:
I think this is a local variable as I'm not seeing it absorbed anywhere.  If that's true, then I believe path needs to be g_free'd according to API doc.

Reply:
Line 405, ` g_autofree char *path = NULL;` same as when the code was introduced.

Reply:
same comments as earlier.